controllers/metal3.io/host_config_data.go

```diff
@@ -94,6 +94,31 @@ func (hcd *hostConfigData) NetworkData() (string, error) {
 	return networkDataRaw, err
 }
 
+// PreprovisioningNetworkData get preprovisioning network configuration
+func (hcd *hostConfigData) PreprovisioningNetworkData() (string, error) {
+	if hcd.host.Spec.PreprovisioningNetworkDataName == "" {
+		hcd.log.Info("PreprovisioningNetworkDataName is not set, returning empty data")
```

Comment:
This may be useful for your debugging now, but it will be recorded on every reconcile. BMO logs are already very chatty, I suggest removing this.

Reply:
It's actually something we already log for all the other *Data functions, but I agree - will remove and propose a follow-up to remove the others 👍

Reply:
The other ones are not called in every state is the difference I think.